diff -ru arch-arm/usr/include/pthread.h sysroot-1.5/usr/include/pthread.h
--- android-ndk-1.6_r1/build/platforms/android-4/arch-arm/usr/include/pthread.h	2009-07-29 10:26:01.000000000 +1000
+++ android-ndk-1.6_r1/build/platforms/android-4/arch-arm/usr/include/pthread.h	2009-05-06 22:14:18.000000000 +1000
@@ -197,6 +197,9 @@
 
 int pthread_getcpuclockid(pthread_t  tid, clockid_t  *clockid);
 
+#include <errno.h>
+static __inline__ int pthread_cancel(pthread_t thread) { return ENOTSUP; }
+
 int pthread_once(pthread_once_t  *once_control, void (*init_routine)(void));
 
 typedef void  (*__pthread_cleanup_func_t)(void*);
diff -ru arch-arm/usr/include/sys/types.h sysroot-1.5/usr/include/sys/types.h
--- android-ndk-1.6_r1/build/platforms/android-4/arch-arm/usr/include/sys/types.h	2009-07-29 10:26:01.000000000 +1000
+++ android-ndk-1.6_r1/build/platforms/android-4/arch-arm/usr/include/sys/types.h	2009-05-06 08:22:14.000000000 +1000
@@ -119,7 +119,9 @@
 typedef uint32_t       u_int32_t;
 typedef uint16_t       u_int16_t;
 typedef uint8_t        u_int8_t;
+#if !defined(__STRICT_ANSI__)
 typedef uint64_t       u_int64_t;
 #endif
+#endif
 
 #endif
