--- mpfr-3.0.0/configure.in~	Thu Jun 10 21:00:14 2010
+++ mpfr-3.0.0/configure.in	Thu Aug 26 17:40:12 2010
@@ -310,6 +310,8 @@
 AC_C_VOLATILE
 MPFR_CONFIGS
 
+AC_CHECK_MEMBERS([struct lconv.decimal_point])
+
 dnl
 dnl Setup GMP detection
 dnl

--- mpfr-3.0.0/mpfr-impl.h~	Thu Jun 10 21:00:14 2010
+++ mpfr-3.0.0/mpfr-impl.h	Thu Aug 26 17:41:58 2010
@@ -1060,7 +1060,11 @@
 #include <locale.h>
 /* Warning! In case of signed char, the value of MPFR_DECIMAL_POINT may
    be negative (the ISO C99 does not seem to forbid negative values). */
+#ifdef HAVE_STRUCT_LCONV_DECIMAL_POINT
 #define MPFR_DECIMAL_POINT (localeconv()->decimal_point[0])
+#else
+#define MPFR_DECIMAL_POINT ((char) '.')
+#endif
 #define MPFR_THOUSANDS_SEPARATOR (localeconv()->thousands_sep[0])
 #else
 #define MPFR_DECIMAL_POINT ((char) '.')

